#  **Nettoyage des données V2** 

In [1]:
import s3fs
import pandas as pd

In [2]:
pd.set_option("display.max_columns", None)

## 1) Récupération des données

In [3]:
from scripts.importation import process_all_years_s3

In [4]:
# Connexion au bucket S3
fs = s3fs.S3FileSystem(client_kwargs={"endpoint_url": "https://minio.lab.sspcloud.fr"})

ROOT_S3_PATH = "mvallat/diffusion/projet"

In [5]:
# Importation bucket S3
MY_BUCKET = "mvallat/diffusion"
FILE_PATH_S3 = f"{MY_BUCKET}/projet/data_brevets.parquet"

with fs.open(FILE_PATH_S3, 'rb') as file_in:
    data_brevets = pd.read_parquet(file_in)

data_brevets.head(10)

doc-number kind country  status publication_country publication_doc-number  \
0    3038352   A1      FR  PUBDEM                  FR                3038352   
1    3038278   A1      FR  PUBDEM                  FR                3038278   
2    3038409   A1      FR  PUBDEM                  FR                3038409   
3    3038425   A1      FR  PUBDEM                  FR                3038425   
4    3038316   A1      FR  PUBDEM                  FR                3038316   
5    3038399   A1      FR  PUBDEM                  FR                3038399   
6    3038381   A1      FR  PUBDEM                  FR                3038381   
7    3038328   A1      FR  PUBDEM                  FR                3038328   
8    3038251   A1      FR  PUBDEM                  FR                3038251   
9    3038374   A1      FR  PUBDEM                  FR                3038374   

  publication_date publication_bopinum publication_nature  \
0         20170106             2017-01             Brevet   
1         20170106             2017-01             Brevet   
2         20170106             2017-01             Brevet   
3         20170106             2017-01             Brevet   
4         20170106             2017-01             Brevet   
5         20170106             2017-01             Brevet   
6         20170106             2017-01             Brevet   
7         20170106             2017-01             Brevet   
8         20170106             2017-01             Brevet   
9         20170106             2017-01             Brevet   

                                     invention-title  \
0  ENSEMBLE POUR PANNEAU DE PORTE ET PANNEAU ASSOCIE   
1  VEHICULE HYBRIDE COMPRENANT UNE SUPPRESSION DE...   
2  PROCEDE DE CONCEPTION ASSISTEE PAR ORDINATEUR ...   
3  DOCUMENT ELECTRONIQUE TEL QU'UNE CARTE A PUCE ...   
4  PROCEDE DE SYNTHESE EN CONTINU D'UN ELASTOMERE...   
5  PROCEDE ET DISPOSITIF DE COMMANDE D'UNE CONSIG...   
6  DISPOSITIF CAPTEUR PREVU POUR ETRE EMBARQUE DE...   
7  VEHICULE FERROVIAIRE POUR LE NETTOYAGE DES FIL...   
8  PLANCHER PVC EN FIBRES COUPEES ET PROCEDE DE F...   
9  INSTALLATION THERMIQUE DE SECHAGE DE MATIERE P...   

                                 applicant_1_orgname  \
0                  FAURECIA INTERIEUR INDUSTRIE None   
1                        IFP ENERGIES NOUVELLES None   
2  CENTRE NATIONAL DE LA RECHERCHE SCIENTIFIQUE None   
3                         OBERTHUR TECHNOLOGIES None   
4  COMPAGNIE GENERALE DES ETABLISSEMENTS MICHELIN...   
5                                    SOFTATHOME None   
6                              BOGUSZEWSKI GUILLAUME   
7                                SNCF MOBILITES None   
8   ZHEJIANG KINGDOM PLASTICS INDUSTRY CO., LTD None   
9                               GROUPE H. LABBE None   

                               applicant_1_address-1       applicant_1_city  \
0                                     2, RUE HENNAPE               NANTERRE   
1                        1 ET 4 AVENUE DE BOIS-PREAU  RUEIL MALMAISON CEDEX   
2                                  3 RUE MICHEL ANGE                  PARIS   
3                         420 RUE D'ESTIENNE D'ORVES               COLOMBES   
4                                    12 COURS SABLON       CLERMONT-FERRAND   
5                           81 AVENUE FRANCOIS ARAGO               NANTERRE   
6         533, RUE DE JUPITER\nBATIMENT B- APPT. B21               JUVIGNAC   
7                                2 PLACE AUX ETOILES            SAINT-DENIS   
8  SOUTH ZONE, CHEZI BRIDGE, HESHAN TOWN,\nZHEJIA...         TONGXIANG CITY   
9           ZAC DE LA TERRE ROUGE RUE DE L'INDUSTRIE        TOURNAN EN BRIE   

  applicant_1_postcode applicant_1_country  \
0                92000                  FR   
1                92852                  FR   
2                75016                  FR   
3                92700                  FR   
4                63000                  FR   
5                92000                  FR   
6                34990 

## 2) Premier tri des demandes de brevet pertinentes

On commence par ...

#### **Suppression de "status" et de "publication_nature"**
Pour notre étude, nous avons choisi de ne pas exploiter les variables "status" et "publication_nature", qui renseignent l’issue de la demande de brevet (délivrance ou non). Leur prise en compte poserait en effet un biais important pour les demandes les plus récentes, qui sont encore en cours d’examen et pourraient être délivrées ultérieurement.

Nous considérons donc les brevets exclusivement comme des demandes de brevet, indépendamment de leur statut final. Ce choix méthodologique nous paraît pertinent dans la mesure où le dépôt d’une demande constitue déjà un indicateur significatif de l’activité inventive et du niveau d’innovation par secteur, qu’elle aboutisse ou non à une délivrance.

In [6]:
data_brevets_clean = data_brevets.drop(columns = ["status", "publication_nature"])

#### **Suppression de "country" et de "publication_country"**

In [7]:
data_brevets_clean["country"].value_counts()

country
FR    112194
NA       101
Name: count, dtype: int64

In [8]:
data_brevets_clean["publication_country"].value_counts()

publication_country
FR    112194
Name: count, dtype: int64

Comme attendu, cette base de données ne contient que des demandes de brevet déposées en France. Nous pouvons donc supprimer ces deux variables.

In [9]:
data_brevets_clean = data_brevets_clean.drop(columns = ["country", "publication_country"])

#### **Sélection en fonction du type de brevet**

Il s'agit d'abord de choisir quel type de brevet nous intéresse, car nos données en contiennent plusieurs types (cf catégorie "kind").

In [10]:
# Colonne "kind"
data_brevets_clean["kind"].value_counts()

kind
A1    107749
A3      4441
NA       102
B1         3
Name: count, dtype: int64

Le document contient trois types de brevets :

- `A1` : demande de brevet  
- `A3` : demande de certificat d'utilité  
- `B1` : brevet délivré  


**Concernant les types B1 (n = 3)** : cette base recensant les demandes de brevet, elle n'est pas censée contenir un brevet B1.  
Probablement dû à une erreur de classification, nous décidons de retirer ces observations.

In [11]:
# Suppression des B1
data_brevets_clean = data_brevets_clean[data_brevets["kind"] != "B1"]

# Vérification qu'on a supprimé 3 observations
data_brevets_clean.shape[0]


112292

Concernant les types A3 (n = 4 441) : les certificats d'utilité sont des certificats délivrés par l'INPI qui protègent l'innovation, tout comme les brevets. En revanche la durée de protection est raccourcie (10 ans contre 20 ans) et la procédure d'examen de la demande est allégée. Pour simplifier l'analyse en la rendant plus homogène, nous ne gardons pas ces demandes A3 dans notre base.

In [12]:
# Suppression des A3
data_brevets_clean = data_brevets_clean[data_brevets_clean["kind"] != "A3"]

# Vérification : suppression de 4 441 observations (112,292 - 4,441 = 107,851)
data_brevets_clean.shape[0]

107851

**Concernant les NA (n = 102)** : plus d'observations sont nécessaires pour décider de la manière de gérer ces NA.


In [13]:
# Observation des NA
brevets_na = data_brevets_clean[data_brevets_clean["kind"] == "NA"]
brevets_na


doc-number kind publication_doc-number publication_date  \
285            NA   NA                   None             None   
606            NA   NA                   None             None   
831            NA   NA                   None             None   
1153           NA   NA                   None             None   
1543           NA   NA                   None             None   
...           ...  ...                    ...              ...   
28236          NA   NA                   None             None   
28556          NA   NA                   None             None   
28760          NA   NA                   None             None   
29096          NA   NA                   None             None   
107329    3145430   NA                3145430         20240802   

       publication_bopinum                                    invention-title  \
285                   None                                                 NA   
606                   None                                                 NA   
831                   None                                                 NA   
1153                  None                                                 NA   
1543                  None                                                 NA   
...                    ...                                                ...   
28236                 None                                                 NA   
28556                 None                                                 NA   
28760                 None                                                 NA   
29096                 None                                                 NA   
107329             2024-31  Surveillance de l’intégrité des données et con...   

       applicant_1_orgname     applicant_1_address-1 applicant_1_city  \
285                     NA                        NA               NA   
606                     NA                        NA               NA   
831                     NA                        NA               NA   
1153                    NA                        NA               NA   
1543                    NA                        NA               NA   
...                    ...                       ...              ...   
28236                   NA                        NA               NA   
28556                   NA                        NA               NA   
28760                   NA                        NA               NA   
29096                   NA                        NA               NA   
107329             AMADEUS  485 ROUTE DU PIN MONTARD             BIOT   

       applicant_1_postcode applicant_1_country applicant_2_orgname  \
285                      NA                  NA                  NA   
606                      NA                  NA                  NA   
831                      NA                  NA                  NA   
1153                     NA                  NA                  NA   
1543                     NA                  NA                  NA   
...                     ...                 ...                 ...   
28236                    NA                  NA                  NA   
28556                    NA                  NA                  NA   
28760                    NA                  NA                  NA   
29096                    NA                  NA                  NA   
107329                06410                  FR                  NA   

       applicant_2_address-1 applicant_2_city applicant_2_postcode  \
285                       NA               NA                   NA   
606                       NA               NA                   NA   
831                       NA               NA                   NA   
1153                      NA               NA                   NA   
1543                      NA               NA                   NA   
...                      ...              ...                  ...   
28236                

In [14]:
# Vérification du nombre d'informations réellement présentes
brevets_na["doc-number"].nunique()
brevets_na["invention-title"].nunique()

2

On peut constater que seules deux observations contiennent des informations. Nous pouvons donc supprimer toutes les observations qui ne contiennent que des NA.

In [15]:
# Suppression des observations 100% NA
data_brevets_clean = data_brevets_clean[data_brevets_clean["kind"] != "NA"]

## 3) Vérification que les observations sont uniques et gestion des observations vides

On commence par vérifier que la base contient bien une observation pour une demande de brevet et qu'il n'y a pas de doublons. 

#### **Unicité du numéro du document**

In [16]:
# Vérification que doc_number est unique

nb_number = data_brevets_clean["doc-number"].nunique()
print("Nombre de doc-number uniques :", nb_number)

nb_number2 = data_brevets_clean["publication_doc-number"].nunique()
print("Nombre de publication_doc-number uniques :", nb_number2)

nb_observations = data_brevets_clean.shape[0]
print("Nombre total d'observations :", nb_observations)


Nombre de doc-number uniques : 107748
Nombre de publication_doc-number uniques : 107748
Nombre total d'observations : 107749


Nous pouvons constater qu'il y a un seul cas de brevets ayant le même numéro. Nous décidons de garder la demande la plus récente en retenant la demande avec le numéro de dossier le plus récent.

In [17]:
# Repérage des doublons
doublons = data_brevets_clean[data_brevets_clean.duplicated(subset=["doc-number"], keep=False)]
print(doublons)

      doc-number kind publication_doc-number publication_date  \
12476    3050837   A1                3050837         20171103   
58919    3050837   A1                3050837         20171103   

      publication_bopinum                                    invention-title  \
12476             2017-44  DISPOSITIF D'INFORMATION INERTIELLE, SAGITTALE...   
58919             2017-44  DISPOSITIF D'INFORMATION INERTIELLE, SAGITTALE...   

                  applicant_1_orgname      applicant_1_address-1  \
12476  BOARDING RING READING SAS None  661 CHE DU CHATEAU VALLON   
58919       BOARDING RING READING SAS  661 CHE DU CHATEAU VALLON   

      applicant_1_city applicant_1_postcode applicant_1_country  \
12476        OLLIOULES                83190                  FR   
58919        OLLIOULES                83190                  FR   

      applicant_2_orgname applicant_2_address-1 applicant_2_city  \
12476                  NA                    NA               NA   
58919               

In [18]:
# Drop de la demande indexée 12476 (on conserve le numéro 58919)
data_brevets_clean = data_brevets_clean.drop(12476)


Enfin, on peut conserver uniquement une des deux variables qui indique le numéro de demande, par exemple "doc-number" et ne pas garder "publication-doc-number".

In [19]:
data_brevets_clean = data_brevets_clean.drop(columns = ["publication_doc-number"])

#### **Absence d'abstract en double**
Les abstracts sont les résumés des brevets déposés. Il s'agit pour nous de vérifier qu'ils contiennent bien du texte d'une part, et qu'il n'y a pas de doublons d'autre part, afin de s'assurer que chaque numéro de demande correspond bien à une demande effective différente.

In [20]:
counts = data_brevets_clean["abstract"].value_counts()
print("Nombre d'abstracts en double :", len(counts[counts > 1]))

Nombre d'abstracts en double : 1166


In [21]:
data_brevets_NA = data_brevets_clean[
    data_brevets_clean["abstract"] == "NA"
]
data_brevets_NA

doc-number kind publication_date publication_bopinum  \
26036    3064421   A1         20180928             2018-39   
26188    3064439   A1         20180928             2018-39   
30080    3068197   A1         20181228             2018-52   
30673    3068791   A1         20190111             2019-02   
31322    3069440   A1         20190201             2019-05   
...          ...  ...              ...                 ...   
50646    3088592   A1         20200522             2020-21   
50648    3088744   A1         20200522             2020-21   
50654    3088523   A1         20200522             2020-21   
50670    3088612   A1         20200522             2020-21   
50708    3088710   A1         20200522             2020-21   

                                         invention-title  \
26036  ROTOR POUR MOTEUR OU GENERATEUR ELECTROMAGNETI...   
26188  DISPOSITIF DE RETROVISION VIDEO POUR VEHICULE ...   
30080  CARTOUCHE DE CARACTERISATION ELECTRONIQUE DES ...   
30673  ANALYSE APPROFONDIE DE L'ORTEQUE DE VERRE OPHT...   
31322                   RENOVATEUR UNIVERSEL DE KERATINE   
...                                                  ...   
50646  DISPOSITIF DE SECURITE POUR CIRCUIT-ELECTRIQUE...   
50648  DISPOSITIF DE DETECTION DE LA PRESENCE DE PERS...   
50654             Chaussures à semelles interchangeables   
50670           Dispositif de récupération d'un véhicule   
50708      ECHANGEUR DE CHALEUR POUR VEHICULE AUTOMOBILE   

              applicant_1_orgname                applicant_1_address-1  \
26036             WHYLOT SAS None        PARC D'ACTIVITES QUERCYPOLE 2   
26188                    DAV None  76 RUE AUGUSTE PERRET - ZI EUROPARC   
30080             SWISSLOG FRANCE                     28 QUAI GALLIENI   
30673            GIROD DENIS PAUL           71 BOULEVARD DE STRASBOURG   
31322  DUPIRE PIERRE JULES JOSEPH               66 RUE VAN DER MEERSCH   
...                           ...                                  ...   
50646                  LIVBAG SAS            2 RUE VILLARET DE JOYEUSE   
50648        INNOVATIVE RESOURCES                  60 ROUTE DU CHATEAU   
50654          OUABBAS ABDERRAHIM                   13 RUE DE LA BUTTE   
50670                 Naval Group          40-42 rue du Docteur Finlay   
50708   VALEO SYSTEMES THERMIQUES    8 RUE LOUIS LORMAND,  LA VERRIERE   

            applicant_1_city applicant_1_postcode applicant_1_country  \
26036                 CAMBES                46100                  FR   
26188          CRETEIL CEDEX                94046                  FR   
30080         SURESNES CEDEX                92156                  FR   
30673               LE HAVRE                76600                  FR   
31322              TOURCOING                59200                  FR   
...                      ...                  ...                 ...   
50646                  PARIS                75017                  FR   
50648              GRISELLES                45210                  FR   
50654                 DRANCY                93700                  FR   
50670                  Paris                75015                  FR   
50708  LE MESNIL SAINT DENIS                78320                  FR   

      applicant_2_orgname applicant_2_address-1 applicant_2_city  \
26036                  NA                    NA               NA   
26188                  NA                    NA               NA   
30080                  NA                    NA               NA   
30673                  NA                    NA               NA   
31322                  NA                    NA               NA   
...                   ...                   ...              ...   
50646                  NA                    NA               NA   
50648                  NA                    NA               NA   
50654                  NA                    NA               NA   
50670                  NA                    NA               NA   
50708                  NA 

- **Gestion des NA** : En observant la base de données, on constate que ces abstract vides sont parfois associés à des demandes incomplètes, et parfois le reste de la demande semble complète. Après quelques recherches sur internet, il nous semble que l'explication la plus probable est que ces demandes soient déposées sans leur abstract, qui est fourni plus tard à l'INPI (l'abstract étant obligatoire pour la demande). Pour se concentrer sur les demandes complètes uniquement, nous décidons donc de supprimer ces observations, qui restent en faible nombre par rapport à nos plus de 100,000 observations.

In [22]:
# Drop des NA
data_brevets_clean = data_brevets_clean[
    data_brevets_clean["abstract"] != "NA"
]

# Vérification 
nb_na = (data_brevets_clean["abstract"] == "NA").sum()
print("Nombre de NA après le drop :", nb_na)

Nombre de NA après le drop : 0


- **Gestion des abstracts en double** : 

In [23]:
# compte des abstracts strictement identiques
duplicates = data_brevets_clean[data_brevets_clean.duplicated(subset="abstract", keep=False)]
print("Nombre d'observations avec des doublons :", len(duplicates))

Nombre d'observations avec des doublons : 2691


On commence par parcourir les observations avec des doublons pour se donner une idée de la raison de l'existence de ces doublons.

In [24]:
duplicates.sort_values("abstract")

doc-number kind publication_date publication_bopinum  \
14977    3053138   A1         20171229             2017-52   
14692    3052641   A1         20171222             2017-51   
74543    3112481   A1         20220121             2022-03   
72807    3110847   A1         20211203             2021-48   
65623    3103688   A1         20210604             2021-22   
...          ...  ...              ...                 ...   
23601    3061660   A1         20180713             2018-28   
9767     3048051   A1         20170825             2017-34   
9851     3048052   A1         20170825             2017-34   
16884    3055060   A1         20180216             2018-07   
16838    3055061   A1         20180216             2018-07   

                                         invention-title  \
14977  PROCEDE D'EDITION AUTOMATISEE D'UN BLOC D'UN T...   
14692  UTILISATION DE XYLITOL POUR AMELIORER LA DISSO...   
74543  COMPOSITION POUR TEINDRE DES FIBRES DE KÉRATIN...   
72807  COMPOSITION POUR TEINDRE DES FIBRES DE KéRATIN...   
65623  Dispositif destiné à assurer la connexion de s...   
...                                                  ...   
23601  COMPOSITIONS SOUS FORME D'UNE SOLUTION AQUEUSE...   
9767   DISPOSITIF DE RACCORDEMENT, PROCEDE ET OUTIL P...   
9851   DISPOSITIF DE RACCORDEMENT, PROCEDE ET OUTIL P...   
16884                         ECRAN MOBILE POUR VEHICULE   
16838                         ECRAN MOBILE POUR VEHICULE   

                                 applicant_1_orgname  \
14977                                LUCKY CART None   
14692                                   EUROTAB None   
74543                                        L'OREAL   
72807                                        L'OREAL   
65623                                  READYNOVATION   
...                                              ...   
23601                                    ADOCIA None   
9767   PARKER HANNIFIN MANUFACTURING FRANCE SAS None   
9851   PARKER HANNIFIN MANUFACTURING FRANCE SAS None   
16884            PEUGEOT CITROEN AUTOMOBILES SA None   
16838            PEUGEOT CITROEN AUTOMOBILES SA None   

                                   applicant_1_address-1     applicant_1_city  \
14977                                 159 RUE MONTMARTRE                PARIS   
14692                                  ZAC DES PEYRARDES   ST JUST ST RAMBERT   
74543                                     14, rue Royale                PARIS   
72807                                     14, rue Royale                PARIS   
65623        9 rue Volta Chez Monsieur Geoffroy GLORIEUX                PARIS   
...                                                  ...                  ...   
23601                              115 AVENUE LACASSAGNE                 LYON   
9767   17 RUE DES BUCHILLONS\nZI DU MONT BLANC - VILL...            ANNEMASSE   
9851   17 RUE DES BUCHILLONS\nZI DU MONT BLANC - VILL...            ANNEMASSE   
16884                                      ROUTE DE GISY  VELIZY VILLACOUBLAY   
16838                                      ROUTE DE GISY  VELIZY VILLACOUBLAY   

      applicant_1_postcode applicant_1_country applicant_2_orgname  \
14977                75002                  FR                  NA   
14692                42170                  FR                  NA   
74543                75008                  FR                  NA   
72807                75008                  FR                  NA   
65623                75003                  FR                  NA   
...                    ...                 ...                 ...   
23601                69003                  FR                  NA   
9767                 74112                  FR                  NA   
9851                 74112                  FR                  NA   
16884                78140                  FR                  NA   
16838                78140                  FR                  NA   

      applicant_2_address-1 applicant_2_city applican

**Cas des doublons parfaits :**

On veut d'abord vérifier si il existe des observations qui ont exactement les mêmes valeurs pour toutes les variables sauf pour le numéro de dossier.

In [25]:
# Vérification existance doublon parfait
cols_comparaison = [
    c for c in data_brevets_clean.columns
    if c not in ["doc_number", "publication_doc_number"]
]


n_duplicates = data_brevets_clean.duplicated(
    subset=cols_comparaison
).sum()
print(n_duplicates)

0


Il n'y a donc aucun doublon parfait ! Toutes les demandes avec le même abstract ne sont pas identiques pour toutes les variables (au-delà du fait que les demandes n'ont évidemment pas le même numéro).

**Gestion des doublons :**

Nous émettons l'hypothèse que les observations avec des NA en double correspondent à la même demande de brevet déposée plusieurs fois afin d'ajouter une correction d'une version à une autre. Ainsi, nous choisissons de considérer que la demande la plus récente entre les doublons est la demande la plus aboutie, et donc la seule que nous garderons. Pour garder la demande la plus récente, on peut utiliser la date de la demande. Dans le cas où la date est la même, il suffit de garder la demande avec le numéro de dossier le plus élevé car ce numéro augmente d'une unité pour chaque nouvelle demande.

In [26]:
# Garder la demande la plus ancienne parmi les doublons

## Passer publication_date au format date
data_brevets_clean["publication_date"] = pd.to_datetime(
    data_brevets_clean["publication_date"],
    format="%Y%m%d",
    errors="coerce"
)
## Trouver la date la plus ancienne par abstract
max_date_per_abstract = (
    duplicates
    .groupby("abstract")["publication_date"]
    .transform("max")
)

## Garder les doublons les plus récents
duplicates_keep = duplicates[
    duplicates["publication_date"] == max_date_per_abstract
]

## Reconstruire la base
data_brevets_clean2 = pd.concat(
    [
        data_brevets_clean[~data_brevets_clean.index.isin(duplicates.index)],
        duplicates_keep
    ],
    axis=0
).sort_index()

## Compter le nouveau nombre de brevets
duplicates2 = data_brevets_clean2[data_brevets_clean2.duplicated(subset="abstract", keep=False)]
print("Nouveau nombre d'observations avec des doublons (drop quand différence de date):", len(duplicates2))


Nouveau nombre d'observations avec des doublons (drop quand différence de date): 1872


In [27]:
# Garder la demande avec le numéro de demande le plus élevée.

## Passer doc-number au format numérique
data_brevets_clean2["doc-number"] = pd.to_numeric(
    data_brevets_clean2["doc-number"],
    errors="coerce"
)

## Trouver le numéro maximal par doublon
max_doc_number = (
    duplicates2
    .groupby("abstract")["doc-number"]
    .transform("max")
)

## Drop les doublons avec le plus petit numéro
duplicates_keep = duplicates2[
    duplicates2["doc-number"] == max_doc_number
]

## Base finale
data_brevets_clean3 = pd.concat(
    [
        data_brevets_clean2[~data_brevets_clean2.index.isin(duplicates2.index)],
        duplicates_keep
    ],
    axis=0
).sort_index()


## Compter le nouveau nombre de brevets
duplicates3 = data_brevets_clean3[data_brevets_clean3.duplicated(subset="abstract", keep=False)]
print("Nouveau nombre d'observations avec des doublons (drop par numéro):", len(duplicates3))


Nouveau nombre d'observations avec des doublons (drop par numéro): 0


In [28]:
print("Nombre d'observations dans la base:", len(data_brevets_clean3))
print("Nombre d'observations avec des abstracts uniques:", data_brevets_clean3["abstract"].nunique())

Nombre d'observations dans la base: 105883
Nombre d'observations avec des abstracts uniques: 105883


## 4) Sélection des variables pertinentes

In [29]:
data_brevets_clean3.head(5)


doc-number kind     publication_date publication_bopinum  \
0    3038352   A1  2017-01-06 00:00:00             2017-01   
1    3038278   A1  2017-01-06 00:00:00             2017-01   
2    3038409   A1  2017-01-06 00:00:00             2017-01   
3    3038425   A1  2017-01-06 00:00:00             2017-01   
4    3038316   A1  2017-01-06 00:00:00             2017-01   

                                     invention-title  \
0  ENSEMBLE POUR PANNEAU DE PORTE ET PANNEAU ASSOCIE   
1  VEHICULE HYBRIDE COMPRENANT UNE SUPPRESSION DE...   
2  PROCEDE DE CONCEPTION ASSISTEE PAR ORDINATEUR ...   
3  DOCUMENT ELECTRONIQUE TEL QU'UNE CARTE A PUCE ...   
4  PROCEDE DE SYNTHESE EN CONTINU D'UN ELASTOMERE...   

                                 applicant_1_orgname  \
0                  FAURECIA INTERIEUR INDUSTRIE None   
1                        IFP ENERGIES NOUVELLES None   
2  CENTRE NATIONAL DE LA RECHERCHE SCIENTIFIQUE None   
3                         OBERTHUR TECHNOLOGIES None   
4  COMPAGNIE GENERALE DES ETABLISSEMENTS MICHELIN...   

         applicant_1_address-1       applicant_1_city applicant_1_postcode  \
0               2, RUE HENNAPE               NANTERRE                92000   
1  1 ET 4 AVENUE DE BOIS-PREAU  RUEIL MALMAISON CEDEX                92852   
2            3 RUE MICHEL ANGE                  PARIS                75016   
3   420 RUE D'ESTIENNE D'ORVES               COLOMBES                92700   
4              12 COURS SABLON       CLERMONT-FERRAND                63000   

  applicant_1_country                        applicant_2_orgname  \
0                  FR                                         NA   
1                  FR                                         NA   
2                  FR            UNIVERSITE D'AIX-MARSEILLE None   
3                  FR                                         NA   
4                  FR  MICHELIN RECHERCHE ET TECHNIQUE S.A. None   

        applicant_2_address-1   applicant_2_city applicant_2_postcode  \
0                          NA                 NA                   NA   
1                          NA                 NA                   NA   
2  58 BOULEVARD CHARLES LIVON  MARSEILLE CEDEX 7                13284   
3                          NA                 NA                   NA   
4      ROUTE LOUIS BRAILLE 10     GRANGES-PACCOT                 1763   

  applicant_2_country               applicant_3_orgname  \
0                  NA                                NA   
1                  NA                                NA   
2                  FR  ECOLE CENTRALE DE MARSEILLE None   
3                  NA                                NA   
4                  CH                                NA   

                               applicant_3_address-1 applicant_3_city  \
0                                                 NA               NA   
1                                                 NA               NA   
2  TECHNOPOLE DE CHATEAU GOMBERT\n38 RUE FREDERIC...        MARSEILLE   
3                                                 NA               NA   
4                                                 NA               NA   

  applicant_3_postcode applicant_3_country inventor_1_last-name  \
0                   NA                  NA           BERTINELLI   
1                   NA                  NA        MILOSAVLJEVIC   
2                13013                  FR               DUPONT   
3                   NA                  NA              GRAGNIC   
4                   NA                  NA               DORATO   

  inventor_1_first-name                               inventor_1_address-1  \
0               RAPHAEL                                 20 RUE D'ECHAVANNE   
1                  MISA                               17 RUE CLAUDE BENARD   
2             GUILLAUME                                   28 RUE DE VERDUN   
3                ELODIE  C/O OBERTHUR TECHNOLOGIES, 420 RUE D'ESTIENNE ...   
4             MARGARITA  MANUFACTURE FRANCAISE DES 

Nous commençons par supprimer quelques variables qui n'apportent aucune nouvelle information, à savoir :
- `kind` car nous n'avons gardé que les brevets de type A1
- `publication_bopinum` qui correspond au numéro de la semaine de la date de publication. Nous conservons uniquement "publication_date" qui donne la même information.
- Les variables au format `application_...` qui contiennent les mêmes informations sur le pays, la date et le numéro de la demande mais cette balise n'était pas présente pour toutes les années, elles contiennent donc beaucoup de valeurs manquantes et n'apportent aucune information nouvelle.
- `date-search-completed`, la date de complétion de la recherche.



Nous supprimons également toutes les variables qui nous semblent peu utiles pour notre sujet et qui ne contiennent pas d'informations intéressantes pour des statistiques descriptives, à savoir :
- Toutes les variables au format `citation_...` qui renseignent sur les "citations" du brevet, c'est-à-dire les références à des documents antérieurs qui sont mentionnés parce qu’ils sont pertinents pour apprécier la nouveauté de l’invention
- `last-fee-payement` et `next-fee-payement` qui donnent les dates de paiement du brevet passées et à venir.


In [30]:
# Drop de ces variables
data_brevets_finale = data_brevets_clean3.drop(columns = ["kind", "publication_bopinum", "date-search-completed", "last-fee-payement", "next-fee-payement",])

data_brevets_finale.drop(columns = data_brevets_finale.filter(regex="^citation").columns, inplace=True)
data_brevets_finale.drop(columns = data_brevets_finale.filter(regex="^application").columns, inplace=True)

data_brevets_finale.head(5)

doc-number     publication_date  \
0    3038352  2017-01-06 00:00:00   
1    3038278  2017-01-06 00:00:00   
2    3038409  2017-01-06 00:00:00   
3    3038425  2017-01-06 00:00:00   
4    3038316  2017-01-06 00:00:00   

                                     invention-title  \
0  ENSEMBLE POUR PANNEAU DE PORTE ET PANNEAU ASSOCIE   
1  VEHICULE HYBRIDE COMPRENANT UNE SUPPRESSION DE...   
2  PROCEDE DE CONCEPTION ASSISTEE PAR ORDINATEUR ...   
3  DOCUMENT ELECTRONIQUE TEL QU'UNE CARTE A PUCE ...   
4  PROCEDE DE SYNTHESE EN CONTINU D'UN ELASTOMERE...   

                                 applicant_1_orgname  \
0                  FAURECIA INTERIEUR INDUSTRIE None   
1                        IFP ENERGIES NOUVELLES None   
2  CENTRE NATIONAL DE LA RECHERCHE SCIENTIFIQUE None   
3                         OBERTHUR TECHNOLOGIES None   
4  COMPAGNIE GENERALE DES ETABLISSEMENTS MICHELIN...   

         applicant_1_address-1       applicant_1_city applicant_1_postcode  \
0               2, RUE HENNAPE               NANTERRE                92000   
1  1 ET 4 AVENUE DE BOIS-PREAU  RUEIL MALMAISON CEDEX                92852   
2            3 RUE MICHEL ANGE                  PARIS                75016   
3   420 RUE D'ESTIENNE D'ORVES               COLOMBES                92700   
4              12 COURS SABLON       CLERMONT-FERRAND                63000   

  applicant_1_country                        applicant_2_orgname  \
0                  FR                                         NA   
1                  FR                                         NA   
2                  FR            UNIVERSITE D'AIX-MARSEILLE None   
3                  FR                                         NA   
4                  FR  MICHELIN RECHERCHE ET TECHNIQUE S.A. None   

        applicant_2_address-1   applicant_2_city applicant_2_postcode  \
0                          NA                 NA                   NA   
1                          NA                 NA                   NA   
2  58 BOULEVARD CHARLES LIVON  MARSEILLE CEDEX 7                13284   
3                          NA                 NA                   NA   
4      ROUTE LOUIS BRAILLE 10     GRANGES-PACCOT                 1763   

  applicant_2_country               applicant_3_orgname  \
0                  NA                                NA   
1                  NA                                NA   
2                  FR  ECOLE CENTRALE DE MARSEILLE None   
3                  NA                                NA   
4                  CH                                NA   

                               applicant_3_address-1 applicant_3_city  \
0                                                 NA               NA   
1                                                 NA               NA   
2  TECHNOPOLE DE CHATEAU GOMBERT\n38 RUE FREDERIC...        MARSEILLE   
3                                                 NA               NA   
4                                                 NA               NA   

  applicant_3_postcode applicant_3_country inventor_1_last-name  \
0                   NA                  NA           BERTINELLI   
1                   NA                  NA        MILOSAVLJEVIC   
2                13013                  FR               DUPONT   
3                   NA                  NA              GRAGNIC   
4                   NA                  NA               DORATO   

  inventor_1_first-name                               inventor_1_address-1  \
0               RAPHAEL                                 20 RUE D'ECHAVANNE   
1                  MISA                               17 RUE CLAUDE BENARD   
2             GUILLAUME                                   28 RUE DE VERDUN   
3                ELODIE  C/O OBERTHUR TECHNOLOGIES, 420 RUE D'ESTIENNE ...   
4             MARGARITA  MANUFACTURE FRANCAISE DES PNEUMATIQUES MICHELI...   

            inventor_1_city inventor_1_postcode inventor_1_country  \
0                 CHENEBIER               70400     

Pour le moment, nous conservons donc les variables suivantes : 
- `doc-number` : numéro de publication du document de brevet.
- `publication_date` : date de publication officielle de la demande de brevet (la date correspond au lundi de la semaine de dépôt)
- `invention-title` : titre de l’invention tel que publié.
- `year` : année de publication, extraite de la date de publication.


- Variables `applicant_n_...` : nom et adresse du n-ième déposant (nous n'avons téléchargé les informations que pour les trois premiers déposants)
- Variables `inventor_n_...` : nom et adresse du n-ième inventeur (nous n'avons téléchargé les informations que pour les trois premiers inventeurs)
- Variables `agent_n_...` : nom et adresse du n-ième mandataire (nous n'avons téléchargé les informations que pour les trois premiers mandataires)
- Variables `owner_n_...` : nom et adresse du n-ième propriétaire (nous n'avons téléchargé les informations que pour les trois premiers propriétaires)

- `classification_1_text`, `classification_2_text`, `classification_3_text` : trois premiers codes de classification internationale des brevets (CIB / IPC), décrivant le domaine technologique de l’invention. Ces sont les variables que nous utiliseront pour mener une analyse sectorielle.

- `abstract` : résumé de l’invention tel que publié.  

**Ces variables forment la base de donnée finale que nous utiliserons pour les premières statistiques descriptives, que nous nommons "data_brevets_finale"**

**De plus, nous créons une base allégée qui ne contient plus les variables pour les déposants, inventeurs, mandataires et propriétaires. Ces variables seront intéressantes pour faire quelques statistiques descriptives mais elles sont nombreuses, ce qui complique l'exploration des données, et ne nous seront pas utiles pour notre partie analyse.**

In [31]:
data_brevets_light = data_brevets_finale.drop(columns=data_brevets_finale.filter(regex="^(applicant|inventor|agent|owner)").columns)

data_brevets_light.head(5)

doc-number     publication_date  \
0    3038352  2017-01-06 00:00:00   
1    3038278  2017-01-06 00:00:00   
2    3038409  2017-01-06 00:00:00   
3    3038425  2017-01-06 00:00:00   
4    3038316  2017-01-06 00:00:00   

                                     invention-title  \
0  ENSEMBLE POUR PANNEAU DE PORTE ET PANNEAU ASSOCIE   
1  VEHICULE HYBRIDE COMPRENANT UNE SUPPRESSION DE...   
2  PROCEDE DE CONCEPTION ASSISTEE PAR ORDINATEUR ...   
3  DOCUMENT ELECTRONIQUE TEL QU'UNE CARTE A PUCE ...   
4  PROCEDE DE SYNTHESE EN CONTINU D'UN ELASTOMERE...   

                               classification_1_text  \
0  F16B   5/07        20060101AFI20150702BHFR    ...   
1  B60W  20/00        20060101AFI20150630BHFR    ...   
2  G06F  17/50        20060101AFI20150701BHFR    ...   
3  G06K  19/077       20060101AFI20150630BHFR    ...   
4  C08C  19/44        20060101AFI20150702BHFR    ...   

                               classification_2_text  \
0  F16B   5/08        20060101ALI20150702BHFR    ...   
1  B60W  10/08        20060101ALI20150630BHFR    ...   
2  G06F  17/11        20060101ALI20150701BHFR    ...   
3  G06K  19/02        20060101ALI20150630BHFR    ...   
4  C08C  19/25        20060101ALI20150702BHFR    ...   

                               classification_3_text  \
0  B60J   5/00        20060101ALI20150702BHFR    ...   
1  B60W  10/115       20120101ALI20150630BHFR    ...   
2  E02B   3/06        20060101ALI20150701BHFR    ...   
3                                                 NA   
4  C08F 297/04        20060101ALI20150702BHFR    ...   

                                            abstract  year  
0  La présente invention concerne un ensemble (10...  2017  
1  Dans un véhicule hybride comprenant un moteur ...  2017  
2  Ce procédé de conception assistée par ordinate...  2017  
3  La fabrication d'un document électronique comp...  2017  
4  L'invention concerne un procédé de synthèse en...  2017